subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DefaultResolvedArtifact.java

```diff
@@ -130,25 +130,19 @@ public File getFile() {
         // This method tries to minimize the number of volatile read/writes.
         // Do NOT try to inline the variables there.
         File f = file;
+
         if (f == null) {
-            synchronized (this) {
-                f = file;
-                if (f == null) {
-                    Throwable err = failure;
-                    if (err != null) {
-                        throw UncheckedException.throwAsUncheckedException(err);
-                    }
-                    try {
-                        f = artifactSource.create();
-                        file = f;
-                    } catch (Throwable e) {
-                        err = e;
-                        failure = err;
-                        throw UncheckedException.throwAsUncheckedException(err);
-                    } finally {
-                        artifactSource = null;
-                    }
-                }
+            Throwable err = failure;
```

Comment:
Is this maybe the source of some of the regression? It looks like the synchronization was introduced in https://github.com/gradle/gradle/commit/013703ab2f119eb6a881e43a951eaab90df2546c to attempt to limit the number of times an artifact was resolved. 

Do you think there's sufficient coverage somewhere to show that we haven't broken this?